haskellPackages.gio & haskellPackages.gtk Fixes

hellwolf opened the pull request:
###### Description of changes

The haskell.packages.ghc94.gio fails to build, affecting packages that relies on it.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hellwolf commented:
hold on, building `.#haskell.packages.ghc94.gtk` still fails... let me see if I can piggyback this in.

cdepillabout commented:
@hellwolf When you figure out your problem, make sure to also base this on the `haskell-updates` branch, not `master`, by doing the "draft -> rebase -> undraft" dance as described in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

hellwolf commented:
> @hellwolf When you figure out your problem, make sure to also base this on the `haskell-updates` branch, not `master`, by doing the "draft -> rebase -> undraft" dance as described in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

Sure done.

I am wondering, if there is already other attempt of fixing the gtk package? It seems an important one?

hellwolf commented:
fwiw:

```
$ nix build .#haskell.packages.ghc94.gtk 
error: builder for '/nix/store/cx9kb62aiaj2mkhq8d3df27ndwwd0c41-gtk-0.15.7.drv' failed with exit code 1;
       last 10 log lines:
       > [2 of 2] Linking Setup
       > configuring
       > configureFlags: --verbose --prefix=/nix/store/nr8lgwjhglxjvwmpbcpzc1805bm7h15c-gtk-0.15.7 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --datadir=/nix/store/lg9fycpx9pmcrwvxpq2k2gw9xxvzgym4-gtk-0.15.7-data/share/ghc-9.4.5 --docdir=/nix/store/rmv7hnlg7lzlr82y2nz9lzrkjhsj5d9k-gtk-0.15.7-doc/share/doc/gtk-0.15.7 --with-gcc=gcc --package-db=/build/tmp.yFVlNFYoqu/package.conf.d --ghc-options=-j12 +RTS -A64M -RTS --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --ghc-options=-haddock --extra-lib-dirs=/nix/store/q5mhssfls6iych80439511vz7539gd95-ncurses-6.4/lib --extra-lib-dirs=/nix/store/zi3wndir89vvbly8fkdwvq9v17vrpw8h-libffi-3.4.4/lib --extra-lib-dirs=/nix/store/q7ax2400mx46afnkcbfxfjddzgl4ck44-gmp-with-cxx-6.2.1/lib --extra-include-dirs=/nix/store/82mg5carpqkrsvkq3mwzm8lzjl7hai2g-gtk+-2.24.33-dev/include --extra-lib-dirs=/nix/store/82mg5carpqkrsvkq3mwzm8lzjl7hai2g-gtk+-2.24.33-dev/lib --extra-include-dirs=/nix/store/39y7z1csp1gwbv4kapxf9kxx3rj29403-glib-2.76.1-dev/include --extra-lib-dirs=/nix/store/39y7z1csp1gwbv4kapxf9kxx3rj29403-glib-2.76.1-dev/lib --extra-include-dirs=/nix/store/lf6arf2qj94qs6bkzfl2y75chb7awmx9-zlib-1.2.13-dev/include --extra-lib-dirs=/nix/store/lf6arf2qj94qs6bkzfl2y75chb7awmx9-zlib-1.2.13-dev/lib --extra-lib-dirs=/nix/store/2ldgh1qis9p4zf8bgsdm7897gw8vv36g-zlib-1.2.13/lib --extra-include-dirs=/nix/store/5nl887dh4mi80dm1m243bzxng81glc8i-libffi-3.4.4-dev/include --extra-lib-dirs=/nix/store/5nl887dh4mi80dm1m243bzxng81glc8i-libffi-3.4.4-dev/lib --extra-lib-dirs=/nix/store/zi3wndir89vvbly8fkdwvq9v17vrpw8h-libffi-3.4.4/lib --extra-include-dirs=/nix/store/rvw19191b3w7z0q99r0qnl5gwp3jr8n2-gettext-0.21/include --extra-lib-dirs=/nix/store/rvw19191b3w7z0q99r0qnl5gwp3jr8n2-gettext-0.21/lib --extra-include-dirs=/nix/store/c4rbjkvlwcicy0z0v85h4lbi6qxk2m31-glibc-iconv-2.37/include --extra-lib-dirs=/nix/store/pdg7ljsallvykhjwdh8193i2s4p3a4k4-glib-2.76.1/lib --extra-include-dirs=/nix/store/byd60fgkicd0fl5fvpds96jx6qq1rf3d-cairo-1.16.0-dev/include --extra-lib-dirs=/nix/store/byd60fgkicd0fl5fvpds96jx6qq1rf3d-cairo-1.16.0-dev/lib --extra-include-dirs=/nix/store/fwx7acy9cjyxirbpni63v5l9qg681skh-fontconfig-2.14.0-dev/include --extra-lib-dirs=/nix/store/fwx7acy9cjyxirbpni63v5l9qg681skh-fontconfig-2.14.0-dev/lib --extra-include-dirs=/nix/store/8k1k5skz60xawzy4v3b5wdazq2ps8r6c-freetype-2.13.0-dev/include --extra-lib-dirs=/nix/store/8k1k5skz60xawzy4v3b5wdazq2ps8r6c-freetype-2.13.0-dev/lib --extra-include-dirs=/nix/store/n0hk7kbmiy9glz7jvx6zycfbbcdq735n-bzip2-1.0.8-dev/include --extra-lib-dirs=/nix/store/n0hk7kbmiy9glz7jvx6zycfbbcdq735n-bzip2-1.0.8-dev/lib --extra-lib-dirs=/nix/store/by4gv37sharnf370ki60zam5s7qklqqh-bzip2-1.0.8/lib --extra-include-dirs=/nix/store/fsl89imqn7glm4wivnl772d2h849rhdc-brotli-1.0.9-dev/include --extra-lib-dirs=/nix/store/fsl89imqn7glm4wivnl772d2h849rhdc-brotli-1.0.9-dev/lib --extra-lib-dirs=/nix/store/9gka2mj1hm3s8m21xln47is1529ir6ny-brotli-1.0.9-lib/lib --extra-include-dirs=/nix/store/gyzxvvpriv3cl40a2b6zaxnwd5xkhlmj-libpng-apng-1.6.39-dev/include --extra-lib-dirs=/nix/store/gyzxvvpriv3cl40a2b6zaxnwd5xkhlmj-libpng-apng-1.6.39-dev/lib --extra-lib-dirs=/nix/store/87ss2z65fh1vn052jlzbp51vwpair2ws-libpng-apng-1.6.39/lib --extra-lib-dirs=/nix/store/qbazz3n06qgvm0ima7rh27mk67yiwdc8-freetype-2.13.0/lib --extra-lib-dirs=/nix/store/5cd778rq3ykjpirdwi16wvblkskhj264-fontconfig-2.14.0-lib/lib --extra-include-dirs=/nix/store/ikvk5h1wy3sc3vaf4ccych0xrkhp6w1p-expat-2.5.0-dev/include --extra-lib-dirs=/nix/store/ikvk5h1wy3sc3vaf4ccych0xrkhp6w1p-expat-2.5.0-dev/lib --extra-lib-dirs=/nix/store/si8ypl8afi2s301wwxrrcvirz05x022r-expat-2.5.0/lib --extra-include-dirs=/nix/store/jcqpapaplsfp91nc656mv44jmbp0m2gl-pixman-0.42.2/include --extra-lib-dirs=/nix/store/jcqpapaplsfp91nc656mv44jmbp0m2gl-pixman-0.42.2/lib --extra-include-dirs=/nix/store/8zzg2w8plb7ax4zjfbyl50wv86dd8bz1-libXext-1.3.4-dev/include --extra-lib-dirs=/nix/store/8zzg2w8plb7ax4zjfbyl50wv86dd8bz1-libXext-1.3.4-dev/lib --extra-include-dirs=/nix/store/qjlpy5db17yzcari5qgrw4wazf63vp0q-xorgproto-2021.5/include --extra-include-dirs=/nix/store/f8ssajgv1qpcibs503jacsfd7wgmww4g-libXau-1.0.9-dev/include --extra-lib-dirs=/nix/store/f8ssajgv1qpcibs503jacsfd7wgmww4g-libXau-1.0.9-dev/lib --extra-lib-dirs=/nix/store/k3ryz9pc069qvd8q4r8pk8jlgycmbvcc-libXau-1.0.9/lib --extra-lib-dirs=/nix/store/f5nqkykqapg20r60f3dad6pn9yb45p37-libXext-1.3.4/lib --extra-include-dirs=/nix/store/snhvz53gwr9acfpwa7yl92qzc0ji6l4k-libXrender-0.9.10-dev/include --extra-lib-dirs=/nix/store/snhvz53gwr9acfpwa7yl92qzc0ji6l4k-libXrender-0.9.10-dev/lib --extra-include-dirs=/nix/store/f525j4v68yl89l93iz0wjq0gkxrhx70z-libX11-1.8.4-dev/include --extra-lib-dirs=/nix/store/f525j4v68yl89l93iz0wjq0gkxrhx70z-libX11-1.8.4-dev/lib --extra-include-dirs=/nix/store/fafw3wb5wvamqaqdjlabqyrypapl1xd8-libxcb-1.14-dev/include --extra-lib-dirs=/nix/store/fafw3wb5wvamqaqdjlabqyrypapl1xd8-libxcb-1.14-dev/lib --extra-lib-dirs=/nix/store/fxdjjfnh9csd8sl6w26c49pbirh0ialr-libxcb-1.14/lib --extra-lib-dirs=/nix/store/xrk3b8djcgivyjy7v9hs0w8abn245382-libX11-1.8.4/lib --extra-lib-dirs=/nix/store/4vfw1575jw6h9srk32zb5r568md3gd7f-libXrender-0.9.10/lib --extra-include-dirs=/nix/store/w13591a7q160s7025a80xdjcbcshq2k0-xcb-util-0.4.0-dev/include --extra-lib-dirs=/nix/store/w13591a7q160s7025a80xdjcbcshq2k0-xcb-util-0.4.0-dev/lib --extra-lib-dirs=/nix/store/245pfnlkah0qdm2w098918pp6rigib1q-xcb-util-0.4.0/lib --extra-include-dirs=/nix/store/sr9gwxbdmpi3lb1bz0v5dvgck95fcv58-libGL-1.6.0-dev/include --extra-lib-dirs=/nix/store/sr9gwxbdmpi3lb1bz0v5dvgck95fcv58-libGL-1.6.0-dev/lib --extra-lib-dirs=/nix/store/qbr2f598v05wn3wkxghk1rvd2ds1rmb0-libGL-1.6.0/lib --extra-lib-dirs=/nix/store/0c35qnpznypfc6jdw26kasdbfz39s3if-libglvnd-1.6.0/lib --extra-include-dirs=/nix/store/whgh6f2vlyv980fbip076wsnjca34r1r-libglvnd-1.6.0-dev/include --extra-lib-dirs=/nix/store/whgh6f2vlyv980fbip076wsnjca34r1r-libglvnd-1.6.0-dev/lib --extra-lib-dirs=/nix/store/6vbjx9vvc1gjbc9mdz6lwrwr7z3px9vz-cairo-1.16.0/lib --extra-include-dirs=/nix/store/j9cgl2qpfjyrgqlf9rbr1i5w2z8b2gqx-pango-1.50.14-dev/include --extra-lib-dirs=/nix/store/j9cgl2qpfjyrgqlf9rbr1i5w2z8b2gqx-pango-1.50.14-dev/lib --extra-include-dirs=/nix/store/8y4q81sliddcdg4kiigvs23f3p1a5axy-harfbuzz-7.1.0-dev/include --extra-lib-dirs=/nix/store/8y4q81sliddcdg4kiigvs23f3p1a5axy-harfbuzz-7.1.0-dev/lib --extra-include-dirs=/nix/store/1qqjmcq93hc7qsshvzyrnc9s1fh5a3q5-graphite2-1.3.14/include --extra-lib-dirs=/nix/store/1qqjmcq93hc7qsshvzyrnc9s1fh5a3q5-graphite2-1.3.14/lib --extra-lib-dirs=/nix/store/cj1992h929iymzvbaj15s4hpli5yb1xq-harfbuzz-7.1.0/lib --extra-include-dirs=/nix/store/fvvw0pw4ypmdchsp1w2bjh5lgchcfqwf-libXft-2.3.6-dev/include --extra-lib-dirs=/nix/store/fvvw0pw4ypmdchsp1w2bjh5lgchcfqwf-libXft-2.3.6-dev/lib --extra-lib-dirs=/nix/store/r045y00m78c5r0d2nx0y7cc61xvlqb4m-libXft-2.3.6/lib --extra-lib-dirs=/nix/store/v47jn4xsjb3wnc1yz5bagigr2brzv36g-pango-1.50.14/lib --extra-include-dirs=/nix/store/4jgr65if55lvw95q62yqapkqim13fcrp-gdk-pixbuf-2.42.10-dev/include --extra-lib-dirs=/nix/store/4jgr65if55lvw95q62yqapkqim13fcrp-gdk-pixbuf-2.42.10-dev/lib --extra-include-dirs=/nix/store/q83r3c2c0ziw2kch7iaygpqw9hq0lbjk-libtiff-4.5.0-dev/include --extra-lib-dirs=/nix/store/q83r3c2c0ziw2kch7iaygpqw9hq0lbjk-libtiff-4.5.0-dev/lib --extra-include-dirs=/nix/store/vl5x6vail0czk8s5s8jz81wwb3l1wdrp-libjpeg-turbo-2.1.4-dev/include --extra-lib-dirs=/nix/store/vl5x6vail0czk8s5s8jz81wwb3l1wdrp-libjpeg-turbo-2.1.4-dev/lib --extra-lib-dirs=/nix/store/qk1bkdd3p3d7cyw97v6q882a3557lwh9-libjpeg-turbo-2.1.4/lib --extra-include-dirs=/nix/store/b76imzrhblhnaardsvcg0id3arykczlk-xz-5.4.2-dev/include --extra-lib-dirs=/nix/store/b76imzrhblhnaardsvcg0id3arykczlk-xz-5.4.2-dev/lib --extra-lib-dirs=/nix/store/5533i5qzsxysjf38p1nlza9g9wvh09wk-xz-5.4.2/lib --extra-lib-dirs=/nix/store/pxhxaw32p49l573mnsswb5dfykjxk3zq-libtiff-4.5.0/lib --extra-lib-dirs=/nix/store/fxz9jb5rks0qjsb0vgz6bpmpd1dk8yqn-gdk-pixbuf-2.42.10/lib --extra-include-dirs=/nix/store/3m8yx39r5g8w5pj21q2yc0js0q1pdsj9-at-spi2-core-2.48.0-dev/include --extra-lib-dirs=/nix/store/3m8yx39r5g8w5pj21q2yc0js0q1pdsj9-at-spi2-core-2.48.0-dev/lib --extra-include-dirs=/nix/store/rcywra0lbhi0716kfk8pbgj2fwi3y38s-dbus-1.14.6-dev/include --extra-lib-dirs=/nix/store/rcywra0lbhi0716kfk8pbgj2fwi3y38s-dbus-1.14.6-dev/lib --extra-lib-dirs=/nix/store/nza07njlr2d8yapmbyp9yvckkm79fsm2-dbus-1.14.6-lib/lib --extra-lib-dirs=/nix/store/w2865s7kqhll9x3d7pn7i9pvzp3yqxn2-dbus-1.14.6/lib --extra-lib-dirs=/nix/store/8ds9pxpx50bvg5r298d81dyxl1vqpav9-at-spi2-core-2.48.0/lib --extra-include-dirs=/nix/store/yy0cxffvvxxk646jnb8qvp9jwik9f5qx-libXrandr-1.5.2-dev/include --extra-lib-dirs=/nix/store/yy0cxffvvxxk646jnb8qvp9jwik9f5qx-libXrandr-1.5.2-dev/lib --extra-lib-dirs=/nix/store/lvwnnpp96r61mzgngh134n19jiifmfh5-libXrandr-1.5.2/lib --extra-include-dirs=/nix/store/x3b9h3kxh4rmasax4fmzz7gj6x58xa26-libXcomposite-0.4.5-dev/include --extra-lib-dirs=/nix/store/x3b9h3kxh4rmasax4fmzz7gj6x58xa26-libXcomposite-0.4.5-dev/lib --extra-include-dirs=/nix/store/h0pd23286hq328h9h9f9dwrvc0qv9kk8-libXfixes-6.0.0-dev/include --extra-lib-dirs=/nix/store/h0pd23286hq328h9h9f9dwrvc0qv9kk8-libXfixes-6.0.0-dev/lib --extra-lib-dirs=/nix/store/r0cam1k92j19aw745pygsmk1hz7fj5v1-libXfixes-6.0.0/lib --extra-lib-dirs=/nix/store/4204zdw9mxzjdy9dsfm2j36ampk0gslm-libXcomposite-0.4.5/lib --extra-include-dirs=/nix/store/n8b68g22dj1r16dkcmxxpikp44162b78-libXi-1.8-dev/include --extra-lib-dirs=/nix/store/n8b68g22dj1r16dkcmxxpikp44162b78-libXi-1.8-dev/lib --extra-lib-dirs=/nix/store/d4fs71sh9jgigbnmxy3wjcf4gbkm37zl-libXi-1.8/lib --extra-include-dirs=/nix/store/zn7i0qiy414vb1anbsv4cfp8zlfb7rf5-libXcursor-1.2.0-dev/include --extra-lib-dirs=/nix/store/zn7i0qiy414vb1anbsv4cfp8zlfb7rf5-libXcursor-1.2.0-dev/lib --extra-lib-dirs=/nix/store/klvj220q7hsgh2jxk9aqa4bv79xhxfw8-libXcursor-1.2.0/lib --extra-include-dirs=/nix/store/a469ccjiwam7013bvbhzcvmk73cn90sx-libXinerama-1.1.4-dev/include --extra-lib-dirs=/nix/store/a469ccjiwam7013bvbhzcvmk73cn90sx-libXinerama-1.1.4-dev/lib --extra-lib-dirs=/nix/store/wjxyrh84l046y26y1i7xvz6nd7z3r48w-libXinerama-1.1.4/lib --extra-include-dirs=/nix/store/kngmlgd46bl7v7lj6b26p24f151gq9bl-cups-2.4.2-dev/include --extra-lib-dirs=/nix/store/kngmlgd46bl7v7lj6b26p24f151gq9bl-cups-2.4.2-dev/lib --extra-include-dirs=/nix/store/xzvwh5cqhya45wr83lv0n15alyrz08pm-gmp-with-cxx-6.2.1-dev/include --extra-lib-dirs=/nix/store/xzvwh5cqhya45wr83lv0n15alyrz08pm-gmp-with-cxx-6.2.1-dev/lib --extra-lib-dirs=/nix/store/q7ax2400mx46afnkcbfxfjddzgl4ck44-gmp-with-cxx-6.2.1/lib --extra-lib-dirs=/nix/store/2jaclr2mrwkwhpgqs7l77k9z38kkvg4l-cups-2.4.2-lib/lib --extra-lib-dirs=/nix/store/fqszskyf5vshwj2nl1f7r5dykvbcxpnd-cups-2.4.2/lib --extra-lib-dirs=/nix/store/g7d4cpds94p4jr9lihnj8njw0a62bq14-gtk+-2.24.33/lib --extra-include-dirs=/nix/store/82mg5carpqkrsvkq3mwzm8lzjl7hai2g-gtk+-2.24.33-dev/include --extra-lib-dirs=/nix/store/82mg5carpqkrsvkq3mwzm8lzjl7hai2g-gtk+-2.24.33-dev/lib
       > Using Parsec parser
       > Configuring gtk-0.15.7...
       > CallStack (from HasCallStack):
       >   withMetadata, called at libraries/Cabal/Cabal/src/Distribution/Simple/Utils.hs:370:14 in Cabal-3.8.1.0:Distribution.Simple.Utils
       > Error: Setup: Encountered missing or private dependencies:
       > text >=0.11.0.6 && <1.3
       >
       For full logs, run 'nix log /nix/store/cx9kb62aiaj2mkhq8d3df27ndwwd0c41-gtk-0.15.7.drv'.

```

hellwolf commented:
Found a way to fix gtk.

sternenseemann commented on pkgs/development/haskell-modules/configuration-nix.nix:
```diff
@@ -260,14 +260,20 @@ self: super: builtins.intersectAttrs super {
 
   # https://github.com/NixOS/cabal2nix/issues/136 and https://github.com/NixOS/cabal2nix/issues/216
-  gio = disableHardening ["fortify"] (addPkgconfigDepend pkgs.glib (addBuildTool self.buildHaskellPackages.gtk2hs-buildtools super.gio));
+  gio = lib.foldr (a : b : a b) super.gio
```
```suggestion
  gio = lib.pipe super.gio
```

sternenseemann commented on pkgs/development/haskell-modules/configuration-nix.nix:
```diff
@@ -260,14 +260,20 @@ self: super: builtins.intersectAttrs super {
+      (addBuildTool self.buildHaskellPackages.gtk2hs-buildtools)
+      (addPkgconfigDepends (with pkgs; [ glib pcre2 util-linux libselinux libsepol pcre ]))
+    ];
+  gtk = lib.foldr (a : b : a b) super.gtk
```
```suggestion
  gtk = lib.pipe super.gtk
```

sternenseemann reviewed (changes requested):
Would be nice if you could split the refactoring and the change into separate commits for review.

Also commit message of `haskellPackages.gio: …` etc. is more appropriate since this change affects all package sets. I also hope you have tested this in other (at least the main) package set(s).

sternenseemann commented:
@ofborg build haskellPackages.gio haskellPackages.gtk

hellwolf committed:
haskellPackages.gio: fix build

- More addPkgconfigDepends
- Tested pkgsets:
  - [x] .#haskell.packages.ghc92.gio
  - [x] .#haskell.packages.ghc94.gio

hellwolf committed:
haskellPackages.gtk: fix build

- More addPkgconfigDepends
- doJailbreak
- Tested pkgsets:
  - [x] .#haskell.packages.ghc92.gtk
  - [x] .#haskell.packages.ghc94.gtk

hellwolf commented:
> Would be nice if you could split the refactoring and the change into separate commits for review.
> 
> Also commit message of `haskellPackages.gio: …` etc. is more appropriate since this change affects all package sets. I also hope you have tested this in other (at least the main) package set(s).

Done.

Separated in two two commit, and all tested with ghc92, ghc94. With ghc96, they fail.

maralorn commented:
This looks good to me, and building `.#haskell.packages.ghc9{2,4}.g{io,tk}` works. 

* You haven’t actually split converting into pipe and adding one override into different commits, like sterni asked.
* You have unneccessary parentheses around doJailbreak
* The jailbreak should have a comment with a link to an upstream issue about the restrictive text bound or a link about the unreleased fix or something. If neccessary you need to open an issue like that.

hellwolf committed:
haskellPackages.gtk: refactor configuration-nix

- using lib.pipe

hellwolf commented:
> This looks good to me, and building `.#haskell.packages.ghc9{2,4}.g{io,tk}` works.
> 
>     * You haven’t actually split converting into pipe and adding one override into different commits, like sterni asked.
> 
>     * You have unneccessary parentheses around doJailbreak
> 
>     * The jailbreak should have a comment with a link to an upstream issue about the restrictive text bound or a link about the unreleased fix or something. If neccessary you need to open an issue like that.

check it out again.

maralorn commented:
Sorry, forgot this earlier, but I have one last change request. The `doJailbreak` does not belong in this file but into configuration-ghc-9.4.x.nix if you move it into a separate override there I will merge this.

hellwolf commented:
> Sorry, forgot this earlier, but I have one last change request. The `doJailbreak` does not belong in this file but into configuration-ghc-9.4.x.nix if you move it into a separate override there I will merge this.

done!

maralorn commented:
Excellent!

maralorn commented:
Just realized that this PR breaks eval on darwin because of the selinux dependency.

hellwolf commented:
oh shoot, would disable selinux for gtk on darwin work?

maralorn commented:
I fixed it with https://github.com/NixOS/nixpkgs/pull/227288/commits/f7e7809ad57126f8f33a2d289b01fedf662ccbf4 but this feels a bit fishy. It’s unclear to me whether all those overrides are needed on all plattforms.

hellwolf commented:
> I fixed it with [f7e7809](https://github.com/NixOS/nixpkgs/commit/f7e7809ad57126f8f33a2d289b01fedf662ccbf4) but this feels a bit fishy. It’s unclear to me whether all those overrides are needed on all plattforms.

Agreed. I had to manually add them because of the builds fail on my x86_64-linux environment.
